### PR TITLE
Misc. Cleanups

### PR DESCRIPTION
**Pull request by nbougalis**

### CODE REVIEW

File: src/ripple/app/tests/Transaction_ordering_test.cpp [hunk 1]
```diff
@@ -30,22 +30,31 @@ struct Transaction_ordering_test : public beast::unit_test::suite
...
 
         {
-            std::mutex m;
             std::unique_lock<std::mutex> lk(m);
             // If stepping through this test in a debugger,
             // make the timeout much longer, or use
             //cv.wait(lk, test);
-            expect(cv.wait_for(lk, 2s, test));
+            expect(cv.wait_for(lk, 2s,
+                [&]() { return coroComplete && test(); }));
```
Comment (vinniefalco) [inline src/ripple/app/tests/Transaction_ordering_test.cpp]: This complexity in a unit test does not thrill me

Comment (nbougalis) [inline src/ripple/app/tests/Transaction_ordering_test.cpp]: Same here. I'm willing to entertain alternatives that don't require massive retooling.

Comment (ximinez) [inline src/ripple/app/tests/Transaction_ordering_test.cpp]: Here ya go. The `submitWait` function predated `JobQueue::rendezvous`.

https://github.com/ximinez/rippled/commit/c4891d3e3e311dd4c3bb52c8337dff33bf8642d0

File: src/ripple/app/tests/Transaction_ordering_test.cpp [hunk 2]
```diff
@@ -144,11 +117,8 @@ struct Transaction_ordering_test : public beast::unit_test::suite
             expect(env.seq(alice) == aliceSequence);
         }
 
-        submitWait(env, tx[0],
-            [&]()
-            {
-                return env.seq(alice) == aliceSequence + 5;
-            });
+        env(tx[0]);
+        env.app().getJobQueue().rendezvous();
```
Comment (ximinez) [inline src/ripple/app/tests/Transaction_ordering_test.cpp]: Wow. Great clean up on this test! Really smart to use `rendezvous()`

:smirk:

**Comment by ximinez** [comment]
Looks good. Builds and tests fine in Windows. It looks like this is one of those cases where we can't help but decrease code coverage, because covered code was removed. :+1:

File: src/ripple/app/tests/Transaction_ordering_test.cpp (hunk 2 above)
Comment (nbougalis) [inline src/ripple/app/tests/Transaction_ordering_test.cpp]: Smart guy @ximinez.

**Comment by HowardHinnant** [comment]
:+1:  Belatedly.  Nice cleanup.

**Comment by nbougalis** [comment]
Merged as f2cdeb7d9a27d692325332bf00c07df27d266e61 and 4a78b343d9453a9a4ba8f5119a4f4a65980603dd

**Pull request closed without merging**